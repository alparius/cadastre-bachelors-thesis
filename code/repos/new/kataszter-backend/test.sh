#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
