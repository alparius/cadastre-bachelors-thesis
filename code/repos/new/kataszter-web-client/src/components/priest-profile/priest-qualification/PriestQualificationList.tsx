import React from "react";
import { Card } from "semantic-ui-react";

import IPriestProfileData from "../../../data/priest/PriestProfileData";
import "../../../static/priestProfile.css";
import ItalicFM from "../../../util/styledMessages/ItalicFM";
import QualificationCard from "./QualificationCard";

interface IPriestProfileQualificationListProps {
    priest: IPriestProfileData;
    editMode: boolean;
    setPriest(newPriest: IPriestProfileData): void;
}

const PriestQualificationList: React.FC<IPriestProfileQualificationListProps> = (props: IPriestProfileQualificationListProps) => {
    const { priest, editMode, setPriest } = props;

    const handleDelete = (index: number) => {
        priest.qualifications.splice(index, 1);
        setPriest({
            ...priest,
            qualifications: priest.qualifications
        });
    };

    const handleChangeOnQualification = (labelName: string, value: any, index: number) => {
        setPriest({
            ...priest,
            qualifications: priest.qualifications.map((qualification, i) =>
                i === index ? { ...qualification, [labelName]: value } : { ...qualification }
            )
        });
    };

    if (!priest.qualifications || priest.qualifications.length === 0) {
        return <ItalicFM id="general.noData" />;
    } else {
        return (
            <Card.Group stackable doubling itemsPerRow={2}>
                {priest.qualifications.map((_, index: number) => (
                    <QualificationCard
                        key={index}
                        qualification={priest.qualifications[index]}
                        index={index}
                        editMode={editMode}
                        handleDelete={handleDelete}
                        handleChangeOnQualification={handleChangeOnQualification}
                    />
                ))}
            </Card.Group>
        );
    }
};

export default PriestQualificationList;
