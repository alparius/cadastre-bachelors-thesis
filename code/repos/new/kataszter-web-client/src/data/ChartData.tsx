interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
