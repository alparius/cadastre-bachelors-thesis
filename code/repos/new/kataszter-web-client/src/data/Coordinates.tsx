interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
