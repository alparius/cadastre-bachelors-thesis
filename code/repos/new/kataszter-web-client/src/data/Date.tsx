interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
