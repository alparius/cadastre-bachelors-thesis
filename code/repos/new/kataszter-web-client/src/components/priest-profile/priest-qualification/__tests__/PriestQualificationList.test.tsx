/*===================
  PriestQualificationList.unit.test.tsx
 ===================*/
import React from "react";

import IPriestProfileData from "../../../../data/priest/PriestProfileData";
import IQualification from "../../../../data/priest/Qualification";
import { NewTestPriestProfileData, NewTestQualification } from "../../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter, mountWithIntl } from "../../../../util/test/test-wrapper";
import PriestQualificationList from "../PriestQualificationList";

const handlers = {
    setPriest: jest.fn()
};

describe("PriestQualificationList element", () => {
    afterEach(() => {
        handlers.setPriest.mockClear();
    });
    it("should render a family child list container when the qualification is not empty", () => {
        const priest: IPriestProfileData = NewTestPriestProfileData;
        const container = createComponentWithIntlAndRouter(<PriestQualificationList priest={priest} editMode={false} {...handlers} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a `no data` when the qualification is empty", () => {
        const priest: IPriestProfileData = NewTestPriestProfileData;
        const emptyQualification: IQualification[] = [];
        priest.qualifications = emptyQualification;

        const container = createComponentWithIntlAndRouter(<PriestQualificationList priest={priest} editMode={false} {...handlers} />).toJSON();
        expect(container).toMatchSnapshot();
    });
    it("should call setPriest if delete button pressed", () => {
        const priest: IPriestProfileData = NewTestPriestProfileData;
        priest.qualifications = [NewTestQualification];
        const container = mountWithIntl(<PriestQualificationList priest={priest} editMode {...handlers} />);

        container
            .find("#deleteButton")
            .at(0)
            .simulate("click");
        expect(handlers.setPriest).toHaveBeenCalledTimes(1);
    });

    it.each([
        [0, "2", "3"],
        [1, "2", "3"],
        [2, "2", "3"]
    ])(
        "should  call handleChangeOnQualification on %d. input. After first change: %s, After second change: %s",
        (inputNr, firstChange, secondChange) => {
            const priest: IPriestProfileData = NewTestPriestProfileData;
            priest.qualifications = [NewTestQualification];
            const container = mountWithIntl(<PriestQualificationList priest={priest} editMode {...handlers} />);
            const input = container.find("input").at(Number(inputNr));
            input.simulate("change", { target: { value: firstChange } });
            input.simulate("change", { target: { value: secondChange } });
            expect(handlers.setPriest).toHaveBeenCalledTimes(2);
        }
    );
});
