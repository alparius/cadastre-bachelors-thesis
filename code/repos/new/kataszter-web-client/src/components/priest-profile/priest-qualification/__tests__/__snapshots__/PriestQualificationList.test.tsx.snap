// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriestQualificationList element should render a \`no data\` when the qualification is empty 1`] = `
<i>
  No data
</i>
`;

exports[`PriestQualificationList element should render a family child list container when the qualification is not empty 1`] = `
<div
  className="ui doubling stackable two cards"
>
  <div
    className="ui fluid card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="content"
      >
        <div
          className="header"
          id="header"
        >
          <h2
            className="ui header"
          >
            Erettsegi bizonyitvany
          </h2>
        </div>
      </div>
      <div
        className="description"
      >
        <div
          className="ui list"
          role="list"
        >
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <form
              className="ui form"
              onSubmit={[Function]}
            >
              <div
                className="inline field"
              >
                <b>
                  Genesis date
                  :
                </b>
                 
                <span
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  2000.06.06
                </span>
              </div>
            </form>
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <form
              className="ui form"
              onSubmit={[Function]}
            >
              <div
                className="inline field"
              >
                <b>
                  Issuing Authority
                  :
                </b>
                 
                Felso Oktatasi Intezmeny
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
